packages/react-client/src/hooks/internal/useTrackManager.ts

```diff
@@ -150,7 +140,10 @@ export const useTrackManager = ({
       }
     } else {
       const newTrack = await startDevice();
-      if (!newTrack) throw Error("Device is unavailable");
+      if (!newTrack) {
+        console.warn(`Failed to start ${type} device.`);
+        return;
```

Comment:
Should this info be somehow passed back to user as return value? I would expect that in production app warnings are removed from code, so there is now way to catch this issue.

Reply:
Now `toggleDevice` and `selectDevice` return a `Promise<undefined | DeviceError>` type.